app/src/main/java/io/github/polysmee/settings/FriendsActivity.java

```diff
@@ -0,0 +1,165 @@
+package io.github.polysmee.settings;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.ArrayAdapter;
+import android.widget.AutoCompleteTextView;
+import android.widget.Button;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import io.github.polysmee.R;
+import io.github.polysmee.database.DatabaseUser;
+import io.github.polysmee.database.User;
+import io.github.polysmee.database.databaselisteners.StringSetValueListener;
+import io.github.polysmee.login.MainUserSingleton;
+
+import androidx.appcompat.app.AlertDialog;
+import androidx.constraintlayout.widget.ConstraintLayout;
+
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+
+public class FriendsActivity extends AppCompatActivity {
+
+
+    private AutoCompleteTextView searchFriend;
+    private Map<String, String> namesToIds;
+    private List<String> allUsers;
+    private Button friendAddButton;
+    private AlertDialog.Builder builder;
+    private LayoutInflater inflater;
+
+    private Map<String, List<View>> idsToFriendEntries;
+    private final Set<String> friendsIds = new HashSet<>();
+    private LinearLayout scrollLayout;
+    private final User user = MainUserSingleton.getInstance();
+    private StringSetValueListener friendsValuesListener;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_friends);
+        attributeSet();
+        friendsValuesListener = friendListener();
+        showFriendList();
+    }
+
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        user.removeFriendsListener(friendsValuesListener);
+    }
+
+    protected void attributeSet(){
```

Comment:
General comment for the entire class : don't forget to add javadoc on any method that isn't private. Here all methods are protected so they should have javadoc. Or you can make them private if they aren't used anywhere else